crates/rspack_binding_values/src/filename.rs

```diff
@@ -0,0 +1,107 @@
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use napi::{
+  bindgen_prelude::{FromNapiValue, ToNapiValue, ValidateNapiValue},
+  Either, JsFunction,
+};
+use rspack_core::{AssetInfo, LocalFilenameFn, PathData};
+use rspack_core::{Filename, FilenameFn};
+use rspack_napi::{threadsafe_function::ThreadsafeFunction, NapiResultExt};
+use serde::Deserialize;
+
+use crate::{JsAssetInfo, PathData as JsPathData};
+
+/// A js filename value. Either a string or a function
+///
+/// The function type is generic. By default the function type is tsfn.
+#[derive(Debug)]
+pub struct JsFilename<F = ThreadsafeFunction<(JsPathData, Option<JsAssetInfo>), String>>(
+  Either<String, F>,
+);
+
+/// A local js filename value. Only valid in the current native call.
+///
+/// Useful as the type of a parameter that is invoked immediately inside the function.
+pub type LocalJsFilename = JsFilename<JsFunction>;
+
+impl From<LocalJsFilename> for Filename<LocalJsFilenameFn> {
+  fn from(value: LocalJsFilename) -> Self {
+    match value.0 {
+      Either::A(template) => Filename::from(template),
+      Either::B(js_func) => Filename::from_fn(LocalJsFilenameFn(js_func)),
+    }
+  }
+}
+impl From<JsFilename> for Filename {
+  fn from(value: JsFilename) -> Self {
+    match value.0 {
+      Either::A(template) => Filename::from(template),
+      Either::B(theadsafe_filename_fn) => {
+        Filename::from_fn(Arc::new(ThreadSafeFilenameFn(theadsafe_filename_fn)))
+      }
+    }
+  }
+}
+
+impl<F: FromNapiValue + ValidateNapiValue> FromNapiValue for JsFilename<F> {
+  unsafe fn from_napi_value(
+    env: napi::sys::napi_env,
+    napi_val: napi::sys::napi_value,
+  ) -> napi::Result<Self> {
+    Ok(Self(Either::from_napi_value(env, napi_val)?))
+  }
+}
+
+impl<F: ToNapiValue> ToNapiValue for JsFilename<F> {
+  unsafe fn to_napi_value(
+    env: napi::sys::napi_env,
+    val: Self,
+  ) -> napi::Result<napi::sys::napi_value> {
+    Either::to_napi_value(env, val.0)
+  }
+}
+
+impl<'de> Deserialize<'de> for JsFilename {
```

Comment:
Would you please remove this? Deserialization seems not necessary anymore. You can remove every deserialize derives above.

Reply:
Deserialization is used here: https://github.com/web-infra-dev/rspack/blob/c6b58310cfe9db5fc6ef2013177f1f8303d1ce47/crates/rspack_testing/src/run_fixture.rs#L20

Reply:
I see... this is supposed to be removed in the future. I will initiate another PR to fix this.